content/bd/0001-intro-dbms.md

```diff
@@ -0,0 +1,253 @@
+  os acessos para cada utilizador.
+
+- **Armazenamento Persistente**
+
+  Quando usamos um SGBD temos a certeza que quando acabamos de executar uma instrução
+  (e.g. inserir, atualizar, apagar, etc.) essa instrução foi executada e guardada
+  persistentemente. Caso haja uma falha imediatamente após a execução desta instrução,
+  temos a garantia que não perdemos estes dados.
+
+- **Administração dos Dados**
+
+  Permite a separação de funções: quem desenvolve a aplicação não necessita de
+  ser a mesma pessoa que administra a base de dados.
+  Assim, pode-se ter um profissional muito mais experiente só focado no SGBD,
+  melhorando o armazenamento e recolha de dados.
+
+- **Acesso Concorrente e Recuperação de Falhas**
+
+  Os SGBDs proporcionam ferramentas (_transactions_) para minimizar o tempo perdido
+  por acessos concorrentes, minimizando também as falhas que podem ocorrer.
+
+- **Redução do Tempo de Desenvolvimento de Aplicações**
+
+  Com todas as ferramentas disponibilizadas pelos SGBDs, assim como linguagens
+  próprias de interrogação, dedicadas a dados (e.g. SQL), o trabalho dos programadores
+  é imensamente facilitado.
+  Estas linguagens dão uma interface de alto nível para os dados, que tal como já
+  vimos, permitem criar várias abstrações que simplificam o desenvolvimento de aplicações.
+
+[**Desvantagens**](color:red)
+
+Embora sejam muitas as vantagens dos SGBDs, existem algumas (embora poucas) desvantagens.
+
+- **Aplicações Muito Complexas**
+
+  Por vezes podemos ter aplicações muito complexas em que não é possível modelar
+  os nossos dados num sistema relacional.
+
+- **Desempenho Inaceitável (e.g. para aplicações de tempo-real)**
+
+  Para algumas aplicações, como uma bolsa, os dados são alterados tão rapidamente
+  que o desempenho dos SGBDs não é suficiente.
+
+- **Falta de Análise Flexível para Certos Tipos de Dados**
+
+  Maioritariamente para texto, em que podemos querer fazer pesquisas avançadas
+  (_full text search_/_fuzzy search_) que a maioria dos SGBDs não suportam.
+
+- **Falta de Necessidade**
+
+  Para aplicações extremamente simples ou com dados maioritariamente imutáveis,
+  pode não fazer sentido a introdução de um SGBD, por mais pequeno que seja
+  o _overhead_ de o fazer hoje em dia.
+  Uma aplicação deste tipo não beneficiaria de um SGBD.
+
+## Modelo ANSI-SPARC
+
+![Arquitetura ANSI-SPARC](./assets/0001-ansi-sparc-architecture.svg)
+
+O aparecimento do [modelo ANSI-SPARC](https://en.wikipedia.org/wiki/ANSI-SPARC_Architecture) foi
+um passo importante na criação dos SGBDs relacionais que conhecemos hoje em dia, visto
+que propôs a separação da base de dados em 3 níveis.
+
+- **Nível Externo (_views_)**: Uma _view_ descreve como os utilizadores vêm os dados.
+  Cada utilizador pode ter uma _view_ diferente que é particular a si mesmo, que pode
+  ser uma reflexão dos dados que lhe são úteis assim como apenas aqueles que tem acesso.
```

Comment:
```suggestion
  Cada utilizador pode ter uma _view_ diferente, particular a si mesmo, que tanto pode
  ser uma reflexão dos dados que lhe são úteis como apenas aqueles que tem acesso.
```
(?) Maybe improve this.

Reply:
Possibly:
```suggestion
  Cada utilizador pode ter uma _view_ diferente, particular a si mesmo, sendo uma
  reflexão dos dados que lhe são úteis e a que simultaneamente tem acesso.
```

Reply:
Either way, this whole statement seems odd to me. Can users only have one view? Is this just an example of a view they can have? Can views be shared by multiple users?

Reply:
It's difficult to word this properly. From my understanding, a user's view is everything they can see, so it's only one. I guess users can "share" a view, but you can just say they have different views that happen to be equal to each other.

From Wikipedia:
> It allows independent customized user views: Each user should be able to access the same data, but have a different customized view of the data. These should be independent: changes to one view should not affect others.

> External Level (User Views): A user's view of the database describes a part of the database that is relevant to a particular user. It excludes irrelevant data as well as data which the user is not authorised to access.

> The external schemas describe the different external views of the data and there may be many external schemas for a given database.